.github/workflows/compound-java-check.yml

```diff
@@ -0,0 +1,147 @@
+      integration-test-enabled:
+        required: false
+        type: boolean
+        default: false
+        description: 'Whether integration test is enabled'
+      integration-test-projects:
+        required: false
+        type: string
+        default: "['.']"
+        description: 'list of projects for running integration tests'
+      integration-test-task:
+        required: false
+        type: string
+        default: 'integrationTest'
+        description: "Gradle task name to run integration test"
+    secrets:
+      nvd-api-key:
+        required: false
+        description: 'API key to access NVD database'
+      cisa-domain:
+        required: false
+        description: 'Cybersecurity and Infrastructure Security Agency domain'
+      cisa-user:
+        required: false
+        description: 'Cybersecurity and Infrastructure Security Agency user'
+      cisa-password:
+        required: false
+        description: 'Cybersecurity and Infrastructure Security Agency password'
+    outputs:
+      owasp-scan-status:
+        value: ${{ jobs.calculate-statuses.outputs.owasp-check-status }}
+      license-check-status:
+        value: ${{ jobs.calculate-statuses.outputs.license-check-status }}
+      integration-test-status:
+        value: ${{ jobs.calculate-statuses.outputs.integration-test-status }}
+      aggregated-status:
+        description: "Aggregated status"
+        value: ${{ jobs.calculate-statuses.outputs.aggregated-status }}
+
+jobs:
+  owasp-scan:
+    name: 'OWASP scan'
+    if: inputs.scanner-enabled
+    uses: th2-net/.github/.github/workflows/owasp-gradle-scan.yml@integration-tests # FIXME: switch to main branch
+    with:
+      runsOn: ${{ inputs.runs-on }}
+      javaVersion: ${{ inputs.java-version }}
+      javaVendor: ${{ inputs.java-vendor }}
+    secrets:
+        nvd-api-key: ${{ secrets.nvd-api-key }}
+        cisa-domain: ${{ secrets.cisa-domain }}
+        cisa-user: ${{ secrets.cisa-user }}
+        cisa-password: ${{ secrets.cisa-password }}
+
+  license-check:
+    name: 'License check'
+    if: inputs.scanner-enabled
+    uses: th2-net/.github/.github/workflows/license_check.yml@main
+    with:
+      runsOn: ${{ inputs.runs-on }}
+      version: ${{ inputs.app-version }}
+
+  integration-test:
+    name: 'Integration tests'
+    if: inputs.integration-test-enabled
+    strategy:
+      matrix:
+       integration-test-project: ${{ fromJson(inputs.integration-test-projects) }}
+    uses: th2-net/.github/.github/workflows/java-integration-test.yml@integration-tests # FIXME: switch to main branch
+    with:
+      runs-on: ${{ inputs.runs-on }}
+      java-version: ${{ inputs.java-version }}
+      java-vendor: ${{ inputs.java-vendor }}
+      project-path: ${{ matrix.integration-test-project }}
+      integration-test-task: ${{ inputs.integration-test-task }}
+
+  calculate-statuses:
+    name: 'Calculate statuses'
+    runs-on: ${{ inputs.runs-on }}
```

Comment:
question: What is the point of configuring the runner here? We always execute the bash script. Let's just use `ubuntu-latest` here

Reply:
Just for tradition )